src/v3/src/util/setUrlQueryParams.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+export const setUrlQueryParams = (
+  url: URL | string,
+  params: Record<string, string> = {},
+) => url + '?' + new URLSearchParams(params).toString();
```

Comment:
https://github.com/okta/okta-signin-widget/pull/3031#discussion_r1087255593
> I found a problem in that this breaks the spec test since the spec test uses okta-verify.html as the dummy url. The URL object constructor does not accept that as a valid url and throws an error:

There's a corner case here where if the `url` already has the `?` in it then this might return an invalid URL, but in practice I don't think we'll run into it.

Using the `URL` class should mitigate this and you might be able to use the `URL(url, base)` overload of the constructor and pass the widget `baseUrl` to it. We have a utility in `settingsUtils` called `getBaseUrl(widgetProps)`, although it looks like it can still return `undefined`.